include/dip.hpp

```diff
@@ -0,0 +1,82 @@
+//===- dip.hpp --------------------------------------------------------===//
```

Comment:
I propose using `*.h` for the consistency and the directory structure like:
```
- include
    - Dialect
    - Target
    - Interface
        - buddy
            - dip.h
```
Maybe we will have a "buddy-c" directory for CAPI and potential python binding.

Reply:
What do yo you think about the following?
```
 - include
      - Dialect
      - Target
      - Interface
          - dip
          - dip.h
```

My reason for the change is that I think `buddy` feels a bit more generic inside `Interface` while `dip` specifies our intent of conveying header files for the `DIP` dialect.

Reply:
As discussed, current directory structure is 
```
- include 
    - Dialect
    - Target
    - Interface
        - buddy
            - dip
                - dip.h
                - memref.h
```